### PR TITLE
AB#421 Separete header rendering

### PR DESCRIPTION
**Pull request by evgenii-nikulin**

### CODE REVIEW

File: src/Feature/Navigation/client/Header/Header/styles.scss [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  position: relative;
+  z-index: 999;
+  background: #666;
+
+  >.toggle {
+    color: #999;
+    text-align: center;
+
+    &:hover {
+      color: #fff;
```
Comment (yuya-yuliya) [inline src/Feature/Navigation/client/Header/Header/styles.scss:62]: Use color variables

File: src/Feature/Navigation/client/Header/NavMenu/styles.scss [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    }
+    i{
+      position: absolute;
+      top: 0;
+      right: 0;
+      height: 40px;
+      width: 40px;
+      text-align: center;
+      line-height: 40px;
+      border-left: 1px solid $second-color-dark;
```
Comment (yuya-yuliya) [inline src/Feature/Navigation/client/Header/NavMenu/styles.scss:33]: Wrong color

File: src/Feature/Navigation/client/Header/NavMenu/styles.scss [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+              margin-top: 5px;
+            }
+            li{
+              text-transform: none;
+            }
+            a{
+              color:#898b8e;
+            }
+            a:hover {
+              color: #111;
```
Comment (yuya-yuliya) [inline src/Feature/Navigation/client/Header/NavMenu/styles.scss:190]: Use colors variables

File: src/Feature/Navigation/client/Header/SearchBox/index.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  bindActionCreators(
+    {
+      ChangeRoute: Context.ChangeRoute,
+    },
+    dispatch,
+  );
+
+const connectedToStore = connect<NavSearchProps>(null, mapDispatchToProps)(SearchBoxComponent);
+
+export const SearchBox = JSS.rendering(connectedToStore);
```
Comment (yuya-yuliya) [inline src/Feature/Navigation/client/Header/SearchBox/index.ts:34]: Use `compose`

File: src/Feature/Navigation/client/Header/UserNav/Component.tsx [hunk 5]
```diff
@@ -91,11 +92,11 @@ export default class NavUser extends Jss.SafePureComponent<NavUserProps, NavUser
...
       <form method="POST" action="/apix/client/commerce/auth/logout" className="sign-out-form">
-        <NavigationLink to="/account" className="btn btn-main">
-          <Jss.Text tag="span" field={{ value: 'My Account', editable: 'My Account' }} />
+        <NavigationLink to="/account" className="btn btn-teal">
+          <JSS.Text tag="span" field={{ value: 'My Account', editable: 'My Account' }} />
         </NavigationLink>
-        <NavigationLink to="/account/order-history" className="btn btn-main no-margin-bottom">
-          <Jss.Text tag="span" field={{ value: 'Order History', editable: 'Order History' }} />
+        <NavigationLink to="/account/order-history" className="btn btn-teal no-margin-bottom">
+          <JSS.Text tag="span" field={{ value: 'Order History', editable: 'Order History' }} />
```
Comment (yuya-yuliya) [inline src/Feature/Navigation/client/Header/UserNav/Component.tsx:99]: Should have `main` postfix class name

File: src/Feature/Navigation/client/Header/UserNav/index.ts [hunk 6]
```diff
@@ -1,57 +1,53 @@
...
-const connectedToStore = connect<GlobalHeaderStateProps, GlobalHeaderDispatchProps, GlobalHeaderOwnProps>(
+const connectedToStore = connect<NavUserStateProps, NavUserDispatchProps>(
   mapStateToProps,
-  mapDispatchToProps
-)(GlobalHeaderComponent);
+  mapDispatchToProps,
+)(UserNavComponent);
 
-export default withExperienceEditorChromes(connectedToStore);
+export const UserNav = JSS.rendering(connectedToStore);
```
Comment (yuya-yuliya) [inline src/Feature/Navigation/client/Header/UserNav/index.ts:53]: Use `compose`

File: src/Feature/Navigation/client/Header/UserNav/styles.scss [hunk 7]
```diff
@@ -111,7 +108,7 @@ nav#nav-user {
     h5 {
       color: $focus-color
     }
-    .sign-up {
+    .forgot-password {
```
Comment (yuya-yuliya) [inline src/Feature/Navigation/client/Header/UserNav/styles.scss:111]: Leave `sign-up` class

**Review by yuya-yuliya — CHANGES_REQUESTED** [review]

**Review by yuya-yuliya — APPROVED** [review]

File: src/Feature/Navigation/client/Header/Header/index.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+        <header className={classnames('main-header', { 'toggle-closed': this.state.isClosed })}>
+          <div className="flex-container">
+            <Placeholder name="header-content" rendering={this.props.rendering} />
+          </div>
+        </header>
+        <header className="mobile-header">
+          <a panel-toggle="panel-toggle" className="toggle" onClick={(e) => this.toggleMenu()}>
+            <i className="fa fa-bars" />
+          </a>
+          <Placeholder name="nav-content" rendering={this.props.rendering} />
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/Header/index.tsx:46]: nav-content => header-content ? navigation-content

File: src/Feature/Navigation/client/Header/HeaderContent/index.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+import * as JSS from 'Foundation/ReactJss/client';
+import * as React from 'react';
+
+import { Placeholder } from '@sitecore-jss/sitecore-jss-react';
+
+import { NavContentProps, NavContentState } from './models';
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/HeaderContent/index.tsx:20]: NavContent => Header Content (??)

File: src/Feature/Navigation/client/Header/HeaderContent/index.tsx [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+import './styles.scss';
+
+class HeaderContentComponent extends JSS.SafePureComponent<NavContentProps, NavContentState> {
+  protected safeRender() {
+    return (
+      <div className="nav-wrap">
+        <div className="wrap-content">
+          <Placeholder name="nav-content" rendering={this.props.rendering} />
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/HeaderContent/index.tsx:28]: header-content

File: src/Feature/Navigation/client/Header/NavMenu/index.tsx [hunk 11]
```diff
@@ -1,57 +1,61 @@
...
 
-import classNames from 'classnames';
+import classnames from 'classnames';
 
-import * as JSS from 'Foundation/ReactJss/client';
+import { Image } from '@sitecore-jss/sitecore-jss-react';
 import { NavigationLink } from 'Foundation/UI/client';
 
-import { NavMainProps, NavMainState } from './models';
+import { NavBarProps, NavBarState } from './models';
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/NavMenu/index.tsx:23]: NavigationMenuProps, NavigationMenuState

File: src/Feature/Navigation/client/Header/NavMenu/index.tsx [hunk 12]
```diff
@@ -1,57 +1,61 @@
...
 import './styles.scss';
 
-export default class NavMain extends JSS.SafePureComponent<NavMainProps, NavMainState> {
-
+export class NavMenuComponent extends JSS.SafePureComponent<NavBarProps, NavBarState> {
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/NavMenu/index.tsx:26]: NavigationMenuComponent

File: src/Feature/Navigation/client/Header/NavMenu/index.tsx [hunk 13]
```diff
@@ -1,57 +1,61 @@
...
     elem.classList.toggle('i-open');
-}
+  }
   protected safeRender() {
-    const { menuItems } = this.props;
+    const { menuItems } = this.props.fields.data.datasource;
 
     return (
-      <nav id="nav-main">
+      <nav className="nav-bar">
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/NavMenu/index.tsx:37]: Why?

File: src/Feature/Navigation/client/Header/NavMenu/models.ts [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+  name: string;
+}
+
+export interface MenuItemDataSource extends JSS.BaseDataSourceItem {
+  title: JSS.GraphQLField<JSS.TextField>;
+  image: JSS.GraphQLField<JSS.ImageField>;
+  commerceCategories: JSS.GraphQLListField<CommerceCategory>;
+}
+
+export interface NavMainDataSource extends JSS.BaseDataSourceItem {
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/NavMenu/models.ts:27]: NavigationMenuDataSource

File: src/Feature/Navigation/client/Header/NavMenu/styles.scss (hunk 3 above)
Comment (larionovadaria) [reply src/Feature/Navigation/client/Header/NavMenu/styles.scss:189]: @evgenii-nikulin

File: src/Feature/Navigation/client/Header/QuickNav/index.tsx [hunk 15]
```diff
@@ -1,34 +1,41 @@
...
-            <nav id="utility">
-                <ul>
-                    <li><a href="">Store Locator</a></li>
-                    <li><a href="">Online Flyer</a></li>
-                    <li><a href="">Language/Currency</a></li>
-                </ul>
-            </nav>
-        );
-    }
+class QuickNavComponent extends JSS.SafePureComponent<NavUtilityProps, NavUtilityState> {
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/QuickNav/index.tsx:21]: Nav => Navigation. Please change such shortening here and everywhere.

File: src/Feature/Navigation/client/Header/QuickNav/models.ts [hunk 16]
```diff
@@ -14,6 +14,6 @@
 
 import * as JSS from 'Foundation/ReactJss/client';
 
-export interface NavUtilityProps {}
+export interface NavUtilityProps extends JSS.Rendering {}
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/QuickNav/models.ts:17]: NavUtility => QuickNavigation

File: src/Feature/Navigation/client/Header/QuickNav/styles.scss [hunk 17]
```diff
@@ -14,45 +14,40 @@
...
 @import 'Foundation/UI/client/common/scss/variables';
 
-nav#utility {
-  position: absolute;
-  top: -40px;
-  right: 190px;
-  @media (max-width: $screen-sm-max){
-    position: static;
-  }
+nav.nav-utility {
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/QuickNav/styles.scss:17]: Please rename class as well

File: src/Feature/Navigation/client/Header/SearchBox/Component.tsx [hunk 18]
```diff
@@ -1,41 +1,40 @@
...
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
 
-import * as React from 'react';
-
 import * as JSS from 'Foundation/ReactJss/client';
+import * as React from 'react';
 
-import { SearchMainProps, SearchMainState } from './models';
+import { NavSearchProps, NavSearchState } from './models';
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/SearchBox/Component.tsx:18]: SearchBox or NavigationSearch

File: src/Feature/Navigation/client/Header/SearchBox/Component.tsx [hunk 19]
```diff
@@ -1,41 +1,40 @@
...
 import './styles.scss';
 
 const SEARCH_INPUT_NAME = 'q';
 
-export default class SearchMain extends JSS.SafePureComponent<SearchMainProps, SearchMainState> {
+export class SearchBoxComponent extends JSS.SafePureComponent<NavSearchProps, NavSearchState> {
```
Comment (larionovadaria) [inline src/Feature/Navigation/client/Header/SearchBox/Component.tsx:23]: SearchBox or NavigationSearch

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/Feature/Navigation/client/Header/HeaderContent/index.tsx (hunk 10 above)
Comment (evgenii-nikulin) [reply src/Feature/Navigation/client/Header/HeaderContent/index.tsx:28]: changed to `navigation-content`

File: src/Feature/Navigation/client/Header/NavMenu/styles.scss (hunk 3 above)
Comment (evgenii-nikulin) [reply src/Feature/Navigation/client/Header/NavMenu/styles.scss:189]: We haven't color for that yet, @yuya-yuliya said leave it like that

File: src/Feature/Navigation/client/Header/NavMenu/index.tsx (hunk 13 above)
Comment (evgenii-nikulin) [reply src/Feature/Navigation/client/Header/NavMenu/index.tsx]: Because navigation menu not unique thing and we don't use id out of css

**Pull request merged**